adminSiteClient/EditorBasicTab.tsx

```diff
@@ -111,7 +114,7 @@ class DimensionSlotView extends React.Component<{
                 this.disposers.push(
                     reaction(
                         () =>
-                            this.grapher.type &&
+                            this.grapher.type ||
```

Comment:
This might've been you playing around, right?
This is a bit of Mobx magic, and I'm not sure making it into `||` is better?

Reply:
This one is intentional but I'm no MobX expert, so I could be off here. Basically, I want `this.updateDefaults()` to run whenever the author adds/removes variables OR updates the chart type. I think, at the moment, `this.updateDefaults()` only runs when a variable is added/removed. So, I changed `&&` to `||` to make sure updating the chart type does indeed trigger an update of the defaults (again, I might be off here – this is more the result of testing than a good understanding of how MobX works).

Reply:
Ah, yeah, this one is very confusing indeed.

Afaict [from the mobx documentation](https://mobx.js.org/reactions.html#reaction), Mobx doesn't care about the result of the function, just the properties that were accessed in it.
For clarity, we should probably simply return `[this.grapher.type, this.grapher.yColumnsFromDimensions.length]`.

Reply:
Ahh, I didn't realise that! I'll update it.

(I'm still a bit confused that `this.grapher.type && this.grapher.yColumnsFromDimensions.length` didn't seem to make `updateDefault()` run when the chart type was changed but maybe this was something else or I misunderstood. I'll give it another try.)